test_dataset.py

```diff
@@ -38,17 +41,9 @@ class FilePartition(NamedTuple):
 class UnittestDataset(Dataset):
```

Comment:
I like the inheritance with the GraularityDatasets subclassing this one. To make things a bit more readable, do you mind not defining `UnittestDataset.__len__()` (just use `pass`) maybe. And then have the actual implementation in `FileGranularityDataset.__len__()`.